### PR TITLE
feat(solana-client): Support SOLANA_METRICS_CONFIG env variable (#2048)

### LINKED ISSUE #2048: Support SOLANA_METRICS_CONFIG

### PR DESCRIPTION
**Pull request by 0xcypher02**
This PR attempts to resolve #2048 by allowing operators to set a `metrics.solana_metrics_config` configuration option which will properly set the SOLANA_METRICS_CONFIG environment variable when spawning the Solana Client.

Please let me know if the base branch should be changed. I know `main` is for firedrancer and not frankendancer, so opted to use the `v0.1` branch as base.

### CODE REVIEW

**Comment by mmcgee-jump** [comment]
Can you squash/rebase the commits instead?

File: src/app/fdctl/config/default.toml [hunk 1]
```diff
@@ -280,6 +280,12 @@ dynamic_port_range = "8900-9000"
     # with the `--rpc-pubsub-enable-vote-subscription` argument.
     pubsub_enable_vote_subscription = false
 
+[metrics]
+    # The Solana Labs client supports a SOLANA_METRICS_CONFIG env variable that exports metrics to an InfluxDB instance.
```
Comment (mmcgee-jump) [inline src/app/fdctl/config/default.toml:284]: Format the comment and wrap like the rest of the file (I think double spaced, 72 chars wide).

File: src/app/fdctl/run/run_solana.c [hunk 2]
```diff
@@ -129,6 +129,11 @@ solana_labs_boot( config_t * config ) {
   if( config->rpc.pubsub_enable_block_subscription ) ADD1( "--rpc-pubsub-enable-block-subscription" );
   if( config->rpc.pubsub_enable_vote_subscription ) ADD1( "--rpc-pubsub-enable-vote-subscription" );
 
+ /* metrics */
+  if( strcmp( config->metrics.solana_metrics_config, "" ) )
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/run_solana.c:133]:
```suggestion
  if( strcmp( config->metrics.solana_metrics_config, "" ) ) {
```

File: src/app/fdctl/run/run_solana.c [hunk 3]
```diff
@@ -129,6 +129,11 @@ solana_labs_boot( config_t * config ) {
...
+     if( FD_UNLIKELY( setenv( "SOLANA_METRICS_CONFIG", config->metrics.solana_metrics_config, 1 ) ) )
+      FD_LOG_ERR(( "setenv() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/run_solana.c:135]:
```suggestion
       FD_LOG_ERR(( "setenv() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
```

**Comment by mmcgee-jump** [comment]
Also please make the commit to `main`, we will cherry pick onto the release branch. Thanks for the contribution!

**Comment by mmcgee-jump** [comment]
Replaced by #2464 , I kept your author attribution but let me know if you want it removed. Thank you for the contribution!

**Comment by 0xcypher02** [comment]
> Replaced by #2464 , I kept your author attribution but let me know if you want it removed. Thank you for the contribution!

Thanks, sorry for not following up here. We've been slammed!

**Pull request closed without merging**